db/external_sst_file_ingestion_job.cc

```diff
@@ -291,24 +291,26 @@ Status ExternalSstFileIngestionJob::GetIngestedFileInfo(
   file_to_ingest->version = DecodeFixed32(version_iter->second.c_str());
 
   auto seqno_iter = uprops.find(ExternalSstFilePropertyNames::kGlobalSeqno);
-  if (file_to_ingest->version == 2) {
-    // version 2 imply that we have global sequence number
-    if (seqno_iter == uprops.end()) {
-      return Status::Corruption(
-          "External file global sequence number not found");
-    }
+  if (file_to_ingest->version == 1 &&
+          (ingestion_options_.allow_blocking_flush == true ||
+           ingestion_options_.allow_global_seqno == true)) {
+    return Status::InvalidArgument("external file version is not supported");
+  }
+
+  if (seqno_iter == uprops.end()) {
+    return Status::Corruption(
+        "External file global sequence number not found");
+  }
 
-    // Set the global sequence number
-    file_to_ingest->original_seqno = DecodeFixed64(seqno_iter->second.c_str());
-    file_to_ingest->global_seqno_offset = props->properties_offsets.at(
-        ExternalSstFilePropertyNames::kGlobalSeqno);
+  // Set the global sequence number
+  file_to_ingest->original_seqno = DecodeFixed64(seqno_iter->second.c_str());
+  file_to_ingest->global_seqno_offset = props->properties_offsets.at(
+      ExternalSstFilePropertyNames::kGlobalSeqno);
 
-    if (file_to_ingest->global_seqno_offset == 0) {
-      return Status::Corruption("Was not able to find file global seqno field");
-    }
-  } else {
-    return Status::InvalidArgument("external file version is not supported");
```

Comment:
we should keep this as well (let's say we are ingesting V3 or V4, we should fail)